src/cmd/pay.rs

```diff
@@ -125,18 +133,43 @@ fn print_txn(
 pub struct Payee {
     address: PublicKey,
     amount: Hnt,
+    memo: u64,
 }
 
 impl FromStr for Payee {
     type Err = Box<dyn std::error::Error>;
 
     fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
-        let pos = s
-            .find('=')
-            .ok_or_else(|| format!("invalid KEY=value: missing `=`  in `{}`", s))?;
-        Ok(Payee {
-            address: s[..pos].parse()?,
-            amount: s[pos + 1..].parse()?,
-        })
+        let mut split = s.split('?');
+
+        if let Some(address) = split.next() {
+            let mut amount = None;
+            let mut memo = 0;
+
+            for segment in split {
+                let pos = segment
+                    .find('=')
+                    .ok_or_else(|| format!("invalid KEY=value: missing `=`  in `{}`", segment))?;
+
+                let key = &segment[..pos];
+                let value = &segment[pos + 1..];
+                match key {
+                    "amount" => amount = Some(value.parse()?),
+                    "memo" => memo = u64::from_b64(value)?,
```

Comment:
So memo is not optional here? Or does the txn define the default memo as 0?

Reply:
No memo _is_ optional. This match basically defines possible keys